staging/src/k8s.io/client-go/rest/watch/decoder_test.go

```diff
@@ -54,7 +54,7 @@ func TestDecoder(t *testing.T) {
 		go func() {
 			data, err := runtime.Encode(scheme.Codecs.LegacyCodec(v1.SchemeGroupVersion), expect)
 			if err != nil {
-				t.Fatalf("Unexpected error %v", err)
+				t.Errorf("Unexpected error %v", err)
```

Comment:
> vendor/k8s.io/client-go/rest/watch/decoder_test.go:54:3: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)

this is not the right fix. A proper fix may require surfacing the error back to the main goroutine through a channel.